hack/ensure-golangci-lint.sh

```diff
@@ -0,0 +1,406 @@
+#!/usr/bin/env bash
```

Comment:
I don't find where this is called/used in the codebase. In case of rke provider it was used to [install](https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/af4782bb2cb3a6492cb082ce90a70f283b0e452a/Makefile#L609-L612) linter

Reply:
It's in Makefile already but the script was missing